### DISCUSSION TITLE
Using Pure.DI with constructor generators

### DISCUSSION THREAD

**Discussion by konrad-gora** [DevTeam/Pure.DI#21] [General]
I have a problem with using Pure.DI and constructors generators. During code generation constructors don't exist, so Pure.DI generates code using the default (empty) ctor. But in the meantime constructor generator generate ctor, so after that, I have compilation errors because Pure.DI code is trying to use not existing actors.
Example constructor generators that I tried:
https://github.com/distantcam/AutoCtor
https://github.com/k94ll13nn3/AutoConstructor

I had a similar problem with generating interfaces, but I could create a partial interface (without any methods) manually so it was enough to make it work.

Does anybody have any ideas about it? Does anybody try to do similar things?
I know, this is not a problem with Pure.DI (I also tried different Compile Time DI), but it is problem with a whole Code Generation (code generators cannot use code from other generators)

**Comment by NikolayPianikov** [comment]
@konrad-gora Pure.DI generates code based on your code in the solution and references to other assemblies. It does not have access to generated code by other code generators. Otherwise, the order of generation would be critical and strange side effects would appear.

**Reply by konrad-gora** [reply]
I understand why it doesn't work, I'm just wondering if there are some workarounds.

**Comment by NikolayPianikov** [comment]
Closed as obsolete
